traffic_ops/traffic_ops_golang/cdni/shared.go

```diff
@@ -0,0 +1,360 @@
+		return []byte(inf.Config.Cdni.JwtDecodingSecret), nil
+	})
+	if err != nil {
+		api.HandleErr(w, r, nil, http.StatusInternalServerError, fmt.Errorf("parsing claims: %w", err), nil)
+		return
+	}
+	if !token.Valid {
+		api.HandleErr(w, r, nil, http.StatusInternalServerError, errors.New("invalid token"), nil)
+		return
+	}
+
+	var expirationFloat float64
+	var ucdn string
+	var dcdn string
+	for key, val := range claims {
+		switch key {
+		case "iss":
+			if _, ok := val.(string); !ok {
+				api.HandleErr(w, r, nil, http.StatusBadRequest, errors.New("invalid token - iss (Issuer) must be a string"), nil)
+				return
+			}
+			ucdn = val.(string)
+		case "aud":
+			if _, ok := val.(string); !ok {
+				api.HandleErr(w, r, nil, http.StatusBadRequest, errors.New("invalid token - aud (Audience) must be a string"), nil)
+				return
+			}
+			dcdn = val.(string)
+		case "exp":
+			if _, ok := val.(float64); !ok {
+				api.HandleErr(w, r, nil, http.StatusBadRequest, errors.New("invalid token - exp (Expiration) must be a float64"), nil)
+				return
+			}
+			expirationFloat = val.(float64)
+		}
+	}
+
+	expiration := int64(expirationFloat)
+
+	if expiration < time.Now().Unix() {
+		api.HandleErr(w, r, nil, http.StatusForbidden, errors.New("token is expired"), nil)
+		return
+	}
+	if dcdn != inf.Config.Cdni.DCdnId {
+		api.HandleErr(w, r, nil, http.StatusForbidden, errors.New("invalid token - incorrect dcdn"), nil)
+		return
+	}
+	if ucdn == "" {
+		api.HandleErr(w, r, nil, http.StatusForbidden, errors.New("invalid token - empty ucdn field"), nil)
+		return
+	}
+
+	capacities, err := getCapacities(inf, ucdn)
+	if err != nil {
+		api.HandleErr(w, r, nil, http.StatusInternalServerError, err, nil)
+		return
+	}
+
+	telemetries, err := getTelemetries(inf, ucdn)
+	if err != nil {
+		api.HandleErr(w, r, nil, http.StatusInternalServerError, err, nil)
+		return
+	}
+
+	fciCaps := Capabilities{}
+	capsList := make([]Capability, 0, len(capacities.Capabilities)+len(telemetries.Capabilities))
+	capsList = append(capsList, capacities.Capabilities...)
+	capsList = append(capsList, telemetries.Capabilities...)
+
+	fciCaps.Capabilities = capsList
+
+	api.WriteRespRaw(w, r, fciCaps)
+}
+
+func getFootprintMap(tx *sql.Tx) (map[int][]Footprint, error) {
+	footRows, err := tx.Query(AllFootprintQuery)
+	if err != nil {
+		return nil, fmt.Errorf("querying footprints: %w", err)
+	}
+	defer log.Close(footRows, "closing foorpint query")
+	footprintMap := map[int][]Footprint{}
+	for footRows.Next() {
+		var footprint Footprint
+		if err := footRows.Scan(&footprint.FootprintType, pq.Array(&footprint.FootprintValue), &footprint.CapabilityId); err != nil {
+			return nil, fmt.Errorf("scanning db rows: %w", err)
+		}
+		val, ok := footprintMap[footprint.CapabilityId]
+		if !ok {
+			footprintMap[footprint.CapabilityId] = []Footprint{footprint}
+		} else {
+			val = append(val, footprint)
+			footprintMap[footprint.CapabilityId] = val
```

Comment:
I believe you might be able to replace this with this:
```
footPrintMap[footprint.CapabilityId] = append(footPrintMap[footprint.CapabilityId], footprint)
```
because `append` can handle null/empty slices properly (no need to initialize them manually first): https://go.dev/play/p/7SUhd7bDEus